dev/webpack.config.js

```diff
@@ -0,0 +1,161 @@
+/* eslint import/no-commonjs: 0 */
+const path = require('path');
+const without = require('lodash/without');
+
+const webpack = require('webpack');
+const ExtractTextPlugin = require('extract-text-webpack-plugin');
+const HtmlWebpackPluging = require('html-webpack-plugin');
+const BabiliPlugin = require('babili-webpack-plugin');
+const AutoDllPlugin = require('autodll-webpack-plugin');
+const HappyPack = require('happypack');
+
+const __DEV__ = process.env.NODE_ENV === 'development';
+const clean = arr => arr.filter(item => item !== false);
+
+module.exports = {
+  devtool: __DEV__ ? 'cheap-eval-source-map' : 'cheap-module-source-map',
+
+  entry: __DEV__
+    ? {
+      bundle: [
+        'webpack-dev-server/client?http://localhost:8080',
+        'webpack/hot/only-dev-server',
+        './dev/app',
+      ],
+      instantsearch: './index.js',
+    }
+    : {
+      bundle: './dev/app',
+    },
+
+  output: {
+    path: path.join(__dirname, 'dist'),
+    filename: '[name].[hash].js',
+    library: '[name]',
+    libraryTarget: 'umd',
+  },
+
+  module: {
+    rules: clean([
```

Comment:
`clean` is not explicit. Why not just filter on a new property that defines if it's dev instead of having `object|false`?